drivers/w5100/w5100.c

```diff
@@ -247,21 +248,29 @@ static int recv(netdev_t *netdev, void *buf, size_t len, void *info)
         uint16_t rp = raddr(dev, S0_RX_RD0, S0_RX_RD1);
         uint16_t psize = raddr(dev, (S0_RX_BASE + (rp & S0_MASK)),
                                   (S0_RX_BASE + ((rp + 1) & S0_MASK)));
-        n = psize - 2;
+        len = psize - 2;
 
-        DEBUG("[w5100] recv: got packet of %i byte (at 0x%04x)\n", n, (int)rp);
+        DEBUG("[w5100] recv: got packet of %i byte (at 0x%04x)\n", len, (int)rp);
 
         /* read the actual data into the given buffer if wanted */
         if (in_buf != NULL) {
+            /* Is provided buffer big enough? */
+            if (len > max_len) {
+                spi_release(dev->p.spi);
+                return -ENOBUFS;
```

Comment:
From docs:
```
If called with buf != NULL and len is smaller than the received packet:
    The received packet is dropped
    The content in buf becomes invalid. (The driver may use the memory to implement the dropping - or may not change it.)
    -ENOBUFS is returned
```

AFAIS, the packet won't be dropped if `len > max_len`. The frame is not removed from the RX buffer

Reply:
Yes, sorry. I created this PR when the exact semantics of that case where still in discussion and I forgot to update this PR to the final decision. (I will check tomorrow the other related PRs as well, as they might have similar issues.)